src/LMS1xx.cpp

```diff
@@ -117,9 +118,24 @@ status_t LMS1xx::queryStatus() {
 
 void LMS1xx::login() {
 	char buf[100];
+	int result;
 	sprintf(buf, "%c%s%c", 0x02, "sMN SetAccessMode 03 F4724744", 0x03);
+	
+	fd_set readset;
+	struct timeval timeout;
+	timeout.tv_sec = 1;
+	timeout.tv_usec = 0;
 
-	write(sockDesc, buf, strlen(buf));
+	do { //loop until data is available to read
+	  ros::Duration(1.0).sleep();
+
+	  write(sockDesc, buf, strlen(buf));
+
+	  FD_ZERO(&readset);
+	  FD_SET(sockDesc, &readset);
+	  result = select(sockDesc + 1, &readset, NULL, NULL, &timeout);
```

Comment:
Is the intent to loop forever here? If so, watch out for the behaviour where it writes back to the timeout var any remaining time. That is to say, perhaps `timeout.tv_sec = 1; timeout.tv_usec = 0;` should be inside rather than outside this loop?